### DISCUSSION TITLE
Remember state of the treeview on the left on exit

### DISCUSSION THREAD

**Discussion by Grisgram** [dbeaver/dbeaver#22532] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Not really a "problem", but a very very awkward task, recurring every time, I open dbeaver.
I have a dbproject with 3 servers, each containing a handful of databases, but in my "dev"-role, i only use one or two catalogues.
Every time i open dbeaver, ALL nodes on the left are collapsed.
**click** project
**click** server1
**click** connections
**click** a database
**click** tables
**click** procedures

that's a lot of clicks just to be able to continue, where I left, and that was only the first server!! it happens TWICE. every time i open the program.

**Describe the solution you'd like**
It would be awesome, if the state of the treeview would be remembered by dbeaver, at least when you create a "project", in the project state files somewhere.
When dbeaver started, it should **try** to open all expanded nodes from the previous session (YES this means auth and all, as a connection gets opened).
Maybe not every user wants that, so it could as well be a setting, as long as the option to remember the tree state somewhere, i'd be happy.

**Describe alternatives you've considered**
there are no alternatives than to accept a double-click-battle everytime I start the program

thanks for reading!
cheers,
Gris

**Comment by LonwoLonwo** [comment, marked as answer]
Hello @Grisgram 

Please look at these settings in the Preferences -> User Interface -> Navigator.

![2024-01-19 10_48_05-Preferences](https://github.com/dbeaver/dbeaver/assets/45152336/7a996d85-2a9d-4521-9b8b-a1f96cd92ccd)

They can probably help in your case.

**Reply by Commifreak** [reply]
That works not the way I want. I enabled both settings (depth: 10) but the navigator does not open the 2rd level folder with its connections. I always have to open the folder with the connections it it. Any ideas?

`Reopen database editors after reconnect` si disabled (so all connections keep closd after restart)

**Comment by Grisgram** [comment]
oh wow -- somehow i missed this setting -- for me it's up to level 7 and it does exactly what I want!

Thanks a thousand times for your kind reply and the screenshot!
my apologies for opening a ticket, that already exists -- i just missed, that this feature already exists
can be closed.

kind regard
Gris

**Comment by LonwoLonwo** [comment]
Thanks for the update @Grisgram
